logger.go

```diff
@@ -74,52 +74,54 @@ func NewLogger(ctx context.Context) *Logger {
 	return l
 }
 
-// ServeHTTP calls the next function and incorporates the response into its response cache.
+// Middleware returns a handler that incorporates the response into its response cache.
```

Comment:
I don't know if this makes sense to change in this review, but we keep referring to the buffer channel as a "cache" and that caused me confusion because I think it's just a processing queue/buffer, right?

Reply:
My intention in this context is that it adds to the cache by way of the buffer. I omitted mention of the buffer, though, because I thought it wouldn't matter from the perspective of a consumer of the Middleware.

I changed the function name `addToCache` to `addToRequestBuffer` to reflect what it's really doing.